File: applications/account-gemfire-rabbit-stream-sink/src/main/java/com/vmare/account/gemfire/rabbit/stream/sink/consumer/AccountConsumer.java
package com.vmare.account.gemfire.rabbit.stream.sink.consumer;

import com.vmare.account.gemfire.rabbit.stream.sink.repository.AccountRepository;
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;


@Component
@RequiredArgsConstructor
public class AccountConsumer implements Consumer<Account> {
    private final AccountRepository repository;

    @Override
    public void accept(Account account) {
        repository.save(account);
    }
}

File: applications/account-geode-amqp-sink/src/main/kotlin/com/vmware/tanzu/data/services/rabbitmq/streaming/account/consumer/AccountConsumer.kt
package com.vmware.tanzu.data.services.rabbitmq.streaming.account.consumer

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.repository.AccountRepository
import org.springframework.stereotype.Component
import java.util.function.Consumer

/**
 * @author Gregory Green
 */
@Component
class AccountConsumer(private val accountService: AccountRepository) :
    Consumer<Account> {
    override fun accept(account: Account) {
        accountService.save(account)
    }
}
